Kitodo-DataManagement/src/main/java/org/kitodo/util/Service.java

```diff
@@ -9,14 +9,27 @@
  * GPL3-License.txt file that was distributed with this source code.
  */
 
-package org.kitodo.forms;
+package org.kitodo.util;
 
+import java.awt.image.RenderedImage;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URI;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.imageio.ImageIO;
+
+import org.kitodo.api.filemanagement.FileManagementInterface;
+import org.kitodo.api.imagemanagement.ImageFileFormat;
+import org.kitodo.api.imagemanagement.ImageManagementInterface;
 import org.kitodo.data.database.beans.Folder;
+import org.kitodo.serviceloader.KitodoServiceLoader;
 
 /**
```

Comment:
I would first ask why here KitodoServiceLoader is added to import. According to project design Data Management module can't access API module.

Reply:
I'm quiet confused. The whole functionality of generation could be implemented outside of Data Management module and Task bean used only for setting. Before this PR Task bean was used correctly as a bean, this PR changes it.

Reply:
Unluckily, not. It was “secretly” used in `CurrentTaskForm` and is directly accessed from JSF files. This did not show up as long as there was a 1:1 correspondence between the `Task` classes’ fields and the input field in the JSF to edit a task. But now there is no such correspondence any more, because one task can generate several folders (0−n). So I need a JSF form bean (the `TaskGenerator` class, now renamed to `Generator`) which is repeatable _n_ times, so I need a separate class for being able to repeat it. I do not see a better idea to do so. Correctly, there should be a separate JSF form bean for the task, but we don’t have yet.